### PR TITLE
add unit test for vineyardctl

### LINKED ISSUE #1301: Add unit test for vineyardctl and vineyard operator
Describe your problem
---------------------

At present, we already have the e2e tests for every important feature of the vineyard operator and the vineyardctl. However, the unit tests for [vineyardctl](https://github.com/v6d-io/v6d/tree/main/k8s/cmd), [controllers](https://github.com/v6d-io/v6d/tree/main/k8s/controllers/k8s) and [webhooks](https://github.com/v6d-io/v6d/tree/main/k8s/pkg/webhook)  are lacking, so the issue mainly contains the following contents.

- [ ] Complete the unit test for [vineyardctl](https://github.com/v6d-io/v6d/tree/main/k8s/cmd) refer to the [doc of writing golang test case](https://go.dev/doc/tutorial/add-a-test).
- [ ] Complete the unit test for [controllers](https://github.com/v6d-io/v6d/tree/main/k8s/controllers/k8s) refer to the [doc of writing controllers test case](https://book.kubebuilder.io/cronjob-tutorial/writing-tests.html)
- [ ] Complete the unit test for [webhooks](https://github.com/v6d-io/v6d/tree/main/k8s/pkg/webhook) refer to the [doc of writing golang test case](https://go.dev/doc/tutorial/add-a-test).

### PR DESCRIPTION
**Pull request by zhuyi1159**
What do these changes do?
-------------------------

This is my first attempt to submit a PR, and I have added a main_test.go test file for vineyardctl  , but there are multiple folders such as 'create' and 'delete' in the cmd directory, and each folder has multiple files. I don't know if I should write a test file for each of these files. I would be very grateful if my doubts can be answered.

Related issue number
--------------------

Fixes #1301

### CODE REVIEW

**Comment by zhuyi1159** [comment]
Thank you for your guidance. Now I have generated corresponding test files (except for the utils subfolder) for almost every go file in the cmd directory by using gotests and resubmitted a commit. Does this comply with regulations? If there are any non-compliance areas, please let me know and I will be very happy to make corrections

**Comment by dashanji** [comment]
The framework looks good to me, but you need to fulfill the test cases in the blank named `// TODO: Add test cases.`.

File: k8s/cmd/commands/create/create_backup_test.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	"github.com/v6d-io/v6d/k8s/apis/k8s/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestNewCreateBackupCmd(t *testing.T) {
+	tests := []struct {
+		name string
+		want *cobra.Command
+	}{
+		// TODO: Add test cases.
```
Comment (dashanji) [inline k8s/cmd/commands/create/create_backup_test.go:32]: Insert your test cases.

**Comment by zhuyi1159** [comment]
I have completed the unit tests for 'create' and 'delete' two folders, and running the 'go test' command shows that they pass. However, because I don't have much knowledge about Go language and Kubernetes, it took me a long time to write the test cases, and I'm not sure if they are compliant even though the 'go test' command passed. Is there any technique for writing test cases? Also, the deadline for submitting the application is approaching, can I submit my GSoC application first?

**Comment by dashanji** [comment]
You'd better to understand what the command will do and then write the unit tests. No problem, just go ahead for OSPP.

**Review by sighingnow — COMMENTED** [review]
Use English for comments.

**Comment by dashanji** [comment]
@zhuyi1159  The update looks better than the original one, but the current design only the default configurations of flags. It's better to add the some test cases for the command itself. There is an example https://github.com/dashanji/v6d/tree/add-unittest

**Pull request closed without merging**